nbgv install adds PackageReference even to vcxproj projects, which breaks their build

Hello,

I have a solution that contains some C++/Cli projects, and doing a ```ngbv install``` breaks ```nuget restore``` when run from the command line.
```
##[error]The nuget command failed with exit code(1) and error(NU1201: Project MyCliProject is not compatible with net462 (.NETFramework,Version=v4.6.2). Project MyCliProject  supports: native (native,Version=v0.0)
```

The issue is that Directory.Build.props adds a ProjectReference for all project types. This fails as the project type is technically "native" for C++/Cli.

I found I can fix this issue my adding a condition to the PackageReference ItemGroup:
```xml
  <ItemGroup Condition="'$(PlatformToolset)' == ''">
    <PackageReference Include="Nerdbank.GitVersioning">
      <Version>3.3.37</Version>
      <PrivateAssets>all</PrivateAssets>
    </PackageReference>
  </ItemGroup>
```

This works since C# projects do not define a PlatformToolset as the C++/Cli projects do.

I would create a PR, but I am unsure if this is a generic enough fix for all supported project types. Your thoughts?

Tristan

--- a/src/nbgv/Program.cs
+++ b/src/nbgv/Program.cs
@@ -8,6 +8,7 @@ namespace Nerdbank.GitVersioning.Tool
     using System.Reflection;
     using System.Threading;
     using System.Threading.Tasks;
+    using Microsoft.Build.Construction;
     using Newtonsoft.Json;
     using NuGet.Common;
     using NuGet.Configuration;
@@ -16,7 +17,6 @@ namespace Nerdbank.GitVersioning.Tool
     using NuGet.Protocol.Core.Types;
     using NuGet.Resolver;
     using Validation;
-    using MSBuild = Microsoft.Build.Evaluation;
 
     internal class Program
     {
@@ -246,18 +246,12 @@ namespace Nerdbank.GitVersioning.Tool
 
             // Create/update the Directory.Build.props file in the directory of the version.json file to add the NB.GV package.
             string directoryBuildPropsPath = Path.Combine(versionJsonRoot, "Directory.Build.props");
-            MSBuild.Project propsFile;
-            if (File.Exists(directoryBuildPropsPath))
-            {
-                propsFile = new MSBuild.Project(directoryBuildPropsPath);
-            }
-            else
-            {
-                propsFile = new MSBuild.Project();
-            }
+            ProjectRootElement propsFile = File.Exists(directoryBuildPropsPath)
+                ? ProjectRootElement.Open(directoryBuildPropsPath)
+                : ProjectRootElement.Create(directoryBuildPropsPath);
 
             const string PackageReferenceItemType = "PackageReference";
-            if (!propsFile.GetItemsByEvaluatedInclude(PackageId).Any(i => i.ItemType == "PackageReference"))
+            if (!propsFile.Items.Any(i => i.ItemType == PackageReferenceItemType && i.Include == PackageId))
             {
                 // Validate given sources
                 foreach (var source in sources)
@@ -279,7 +273,7 @@ namespace Nerdbank.GitVersioning.Tool
                     return ExitCodes.PackageIdNotFound;
                 }
 
-                propsFile.AddItem(
+                var item = propsFile.AddItem(
                     PackageReferenceItemType,
                     PackageId,
                     new Dictionary<string, string>
@@ -287,6 +281,7 @@ namespace Nerdbank.GitVersioning.Tool
                         { "Version", packageVersion },
                         { "PrivateAssets", "all" },
                     });
+                item.Condition = " '$(PlatformToolset)' == '' ";
 
                 propsFile.Save(directoryBuildPropsPath);
             }

--- a/src/nbgv/Program.cs
+++ b/src/nbgv/Program.cs
@@ -281,7 +281,7 @@ namespace Nerdbank.GitVersioning.Tool
                         { "Version", packageVersion },
                         { "PrivateAssets", "all" },
                     });
-                item.Condition = " '$(PlatformToolset)' == '' ";
+                item.Condition = "!Exists('packages.config')";
 
                 propsFile.Save(directoryBuildPropsPath);
             }

